### ISSUE TITLE
Connection to my own server doesn't work

### ISSUE THREAD

**Issue by fkooman** [pia-foss/tunnel-apple#47]
# Summary

I'm trying the Demo application against my own VPN server, but it won't connect because of "unknown protocol" error, this is from the server log:

```Jul 18 14:36:18 vpn.tuxed.net openvpn[26865]: 91.64.X.Y OpenSSL: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol```

Initially my server had:
```
tls-cipher TLS-ECDHE-RSA-WITH-AES-256-GCM-SHA384
tls-version-min 1.2
```
Removing those configuration lines doesn't help, but even with those I expect the Demo app to connect after a quick review of `PIATunnel/Sources/Core/TLSBox.m`.

Is there anything required in the server configuration to make connections work?

**Comment by fkooman** [comment]
The branch I'm working from: https://github.com/fkooman/tunnel-apple/tree/tuxed

**Comment by keeshux** [comment]
@fkooman where you see this line:

https://github.com/fkooman/tunnel-apple/blob/tuxed/Demo/BasicTunnel-iOS/ViewController.swift#L137

you should replace `.pia` with `.vanilla` to stick with the standard OpenVPN protocol. I think I forgot to update the README accordingly.

Let me know if it fixes your issue.

**Comment by keeshux** [comment]
FYI: introduced in #28 and #30.

**Comment by fkooman** [comment]
Thanks for your help! I moved on to the next error ;-) 

```
Jul 19 16:47:55 vpn.tuxed.net openvpn[30618]: WRRWWWRWRRRR91.64.107.150 OpenSSL: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca
Jul 19 16:47:55 vpn.tuxed.net openvpn[30618]: 91.64.107.150 OpenSSL: error:140940E5:SSL routines:ssl3_read_bytes:ssl handshake failure
```

This would indicate that there is an attempt to send a client certificate by PIA, but it fails? I was under the impression that client certificates were (not yet) supported, that is what I wanted to look into :)

If I try with OpenVPN with a client configuration without `<cert>` and `<key>` and user user/pass authentication it does work with this server.

If I modify something in the PIATunnel project, e.g. TLSBox.m`, do I need to run `pod install` or `pod update` in the Demo directory again to get the changes? I'm still trying to get the hang of Xcode :-)

**Comment by keeshux** [comment]
@fkooman client certificates are a different thing, this relates to the CA alerted by the server -at least based on the error message. The default `Handshake` enum values are all bound to the bundled PIA certificates, you must provide your own CA instead.

I see you patched the enum, but I introduced in #31 a way to avoid replacing the bundled PEM files. Again, the README needs updating. Try this:

```
builder.handshake = .custom
builder.ca = "your-ca-pem-as-a-string"
```
